cedar-policy-core/src/parser/cst_to_ast.rs

```diff
@@ -523,35 +540,50 @@ impl cst::PolicyImpl {
     ) -> Result<(PrincipalConstraint, ActionConstraint, ResourceConstraint)> {
         // Tracks where the last variable in the scope ended. We'll point to
         // this position to indicate where to fill in vars if we're missing one.
-        let mut end_of_last_var = self.effect.loc.end();
+        let mut end_of_last_var = self.effect.loc.as_ref().map(|loc| loc.end());
 
         let mut vars = self.variables.iter();
         let maybe_principal = if let Some(scope1) = vars.next() {
-            end_of_last_var = scope1.loc.end();
-            scope1.to_principal_constraint(TolerantAstSetting::Tolerant)
+            end_of_last_var = scope1.loc.as_ref().map(|loc| loc.end()).or(end_of_last_var);
+            scope1.to_principal_constraint(TolerantAstSetting::NotTolerant)
```

Comment:
curious why the change from `Tolerant` to `NotTolerant` (also a few places below)

Reply:
This was a copy-paste mistake 😢